contracts/protocol/tokenization/base/ScaledBalanceTokenBase.sol

```diff
@@ -125,4 +126,42 @@ abstract contract ScaledBalanceTokenBase is MintableIncentivizedERC20, IScaledBa
       emit Burn(user, target, amountToBurn, balanceIncrease, index);
     }
   }
+
+  /**
+   * @notice Implements the basic logic to transfer scaled balance tokens between two users
+   * @dev It emits a mint event with the interest accrued per user
+   * @param sender The source address
+   * @param recipient The destination address
+   * @param amount The amount getting transferred
+   * @param index The next liquidity index of the reserve
+   **/
+  function _transfer(
+    address sender,
+    address recipient,
+    uint256 amount,
+    uint256 index
+  ) internal {
+    uint256 senderScaledBalance = super.balanceOf(sender);
+    uint256 senderBalanceIncrease = senderScaledBalance.rayMul(index) -
+      senderScaledBalance.rayMul(_userState[sender].additionalData);
```

Comment:
@miguelmtzinf  Given that this code operates on `_userState[sender]` (or recipient) after and being 1 slot packed with `additionalData`, is not more gas-efficient to do a 

`UserState memory senderState = _userState[sender];`

and then just operate with the fields in memory?
Unless the compiler takes this into account already

Reply:
Not worth to do what you suggest because we need to update storage sender/recipient 's state later on